core/src/trezor/ui/layouts/tr/recovery.py

```diff
@@ -7,15 +7,10 @@
 from ..common import button_request, interact
 from . import RustLayout, raise_if_not_confirmed, show_warning
 
-if TYPE_CHECKING:
-    from trezor.wire import GenericContext
-    from typing import Iterable, Callable
 
-
-async def request_word_count(ctx: GenericContext, dry_run: bool) -> int:
-    await button_request(ctx, "word_count", code=ButtonRequestType.MnemonicWordCount)
+async def request_word_count(dry_run: bool) -> int:
+    await button_request("word_count", code=ButtonRequestType.MnemonicWordCount)
     count = await interact(
```

Comment:
Do we send two duplicate button requests here - one in `button_request` and second in `interact`?

Reply:
right, kind of an entrenched assumption in the tests too. kinda surprising that we didn't notice this before.

n.b., what is this about? https://github.com/trezor/trezor-firmware/blob/matejcik/get-context/tests/input_flows.py#L1810C34-L1810C34
looks like the warning about wrong word is just up and missing

Reply:
81305badef3cbadaa7de0b696ce83953dae4089f

note that UI tests are failing on the above. I didn't want to just record over the test before we have an idea of what's going on there

Reply:
Yes, I am missing the warning screen locally as well. It is connected with `TR` recovery, I would fix it in the appropriate PR - https://github.com/trezor/trezor-firmware/pull/3119 not to slow down this PR (feel free to revert these changes)